### PR TITLE
Add registry fieldset

### LINKED ISSUE #671: Windows registry fieldset
We need to add a fieldset for Windows registry event. Both Sysmon/winlogbeat and Endgame/Elastic endpoint can collect these. I'm less focused on the creation of registry keys, but thinking more about the creation/modification of values. For more background on how the registry works, here's the [wikipedia page](https://en.wikipedia.org/wiki/Windows_Registry#Structure).

Here's an example event collected from Endgame
```json
{
  "bytes_written": "QwA6AFwAcgB0AGEAXAByAGUAZABfAHQAdABwAFwAYgBpAG4AXABtAHkAYQBwAHAALgBlAHgAZQAAAA==",
  "bytes_written_count": 58,
  "bytes_written_string": "C:\\rta\\red_ttp\\bin\\myapp.exe",
  "event_subtype_full": "registry_modify_event",
  "event_type_full": "registry_event",
  "hostname": "RTA-TESTING",
  "key_path": "\\REGISTRY\\MACHINE\\SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion\\Image File Execution Options\\normalprogram.exe\\Debugger",
  "key_type": "sz",
  "opcode": 1,
  "pid": 9108,
  "process_name": "python.exe",
  "process_path": "C:\\Python27\\python.exe",
  "serial_event_id": 77803,
  "timestamp": 131762384629155140,
  "timestamp_utc": "2018-07-16 18:14:22Z",
  "unique_pid": 77788,
  "user_name": "tester",
  "user_sid": "S-1-5-21-4215045029-3277270250-148079304-1004"
}
```

Some of the keys worth adding to the field set:
- `hive`: If we stick with the abbreviated names, the values are HKLM, HKCC, HKCR, HKCU, HKU. if we go unabbreviated names, this means HKEY_LOCAL_MACHINE, HKEY_CURRENT_CONFIG, etc. in the above example, this is `HKLM` / `HKEY_LOCAL_MACHINE`
- `key`: the hive relative path to the key. this doesn't include the value name. this is somewhat analogous to a directory path in na file system. in the above example this is `SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion\\Image File Execution Options\\normalprogram.exe`
- `value`: the name of the value written to. this is somewhat analogous to a file name. in the above example, this is Debugger
- `path`: the full path, including hive, key, and value. essentially `key_path` in the above example

It also may be worth nesting a field set at `registry.data` which contains
- `type`: the representation of the data. basically tells you how to interpret the bytes written. possible values are: REG_NONE, REG_SZ, REG_EXPAND_SZ, REG_DWORD, etc. in the above example, this is REG_SZ.
- `int_value`: open for naming suggestions, but we could store (+/- careful casting) DWORD/QWORD as i64, or `long` within ES
- `string_values`: type: `array[string]` again, open for a better name, but this would store the strings for any string types -- REG_SZ, REG_EXPAND_SZ, REG_LINNK, REG_MULTI_SZ. in the above example, we would store `["C:\\rta\\red_ttp\\bin\\myapp.exe"]`
- `num_strings`: type: long. this would be the number of strings for REG_MULTI_SZ. I'm not sure how often this would need to be queried over, so this could be derived with `length(string_values)`
- `original_bytes`: type string - I'm not sure the best representation, but this could be base64 to store arbitrary bytes without worrying about unicode. this would be most used for REG_BINARY, but could be stored for any data type.

### PR DESCRIPTION
**Pull request by rw-access**
Added field set for Windows registry.

Closes #671

### CODE REVIEW

**Comment by webmat** [comment]
One more thing. The sorting on the field reference page is by short name, not by title. Which means "Windows Registry" is in the middle of the pack, instead of at the end 😂 

For now simplest would be to rename title to "Registry", and omit Windows, IMO. Or if people feel like the word "Windows" is very important, perhaps we can do "Registry (Windows)".

I prefer that over changing the sorting, as the underlying field set will actually be "registry.*". I feel it's important that the titles follow this closely.

File: schemas/registry.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
+---
+- name: registry
+  title: Windows Registry
+  group: 2
+  short: None
+  description: Fields related to Windows Registry operations.
```
[the remark this answers is not in the document]
Comment (rw-access) [reply schemas/registry.yml]: Oh, also I copied/pasted `group`. What does this value mean? Do we need to dedup it as we merge PRs?

**Comment by rw-access** [comment]
Yes, I'll add back `registry.data.original`, but mark it as **extended**. I think it's fair to make it optional, at the very least to keep storage low. The change to `Registry (Windows)` is probably the most clear, so I'll give that a try.

**Comment by webmat** [comment]
The way I see these potentially big fields is that users always have the power to remove them when & how they see fit. In doing that, they obviously lose the ability to utilize them later on. Elastic can even decide to make this optional & off by default in the solutions.

But in defining the schema itself, it's still useful to define the fields regardless of disk usage. Just like we did for http request/response body.

Perhaps we should add general documentation around this at some point.

File: schemas/registry.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (webmat) [reply schemas/registry.yml]: This is a leftover from these YAML files initially being based on the Beats field defs. Not used by ECS. This is something that needs to be cleaned up. `group:2` is the default and is fine :-)

**Comment by webmat** [comment]
@rw-access Do you think you'll have time to add `registry.data.original` soon? I'd like the descriptions for the data fields to be centered around it. E.g. 1) always store the base64 in original 2) copy to .integer when appropriate, copy to .strings when appropriate.

If you don't think you'll have time, we can still proceed and add `.original` later.

But we still need to fix the following before merging:

* the field set's `short: None`
* change the title to "Registry (Windows)" or equivalent, to avoid sorting surprises

File: schemas/registry.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      type: keyword
+      short: Original bytes written with base64 encoding.
+      example: 0
+      description: >
+        Original bytes written with base64 encoding.
+
+        For Windows registry operations, such as SetValueEx and RegQueryValueEx,
+        this corresponds to the data pointed by `lp_data`. This is optional and may be redundant
+        when `data.integer` or data.strings` are populated, but should be populated
+        for REG_BINARY encoded values.
```
Comment (webmat) [inline schemas/registry.yml]: I would have preferred to establish this as a value that's always reliably populated (not optional), and appropriate registry types duplicate the value in a more consumable fashion (.integer for REG_DWORD and REG_QWORD, same for strings).

However I don't see this as a blocker for the PR. Reinforcing the requirement in this direction will not be a breaking change.

What do you think? If you're not convinced of the value we can revisit this later and merge this.

**Comment by webmat** [comment]
As discussed out of band, we're removing `registry.data.integer` for now, since `long` in ES is signed, and in the registry is unsigned.

We'll move forward and merge the rest of this PR as is, and continue thinking on what we do with the `.integer` field. The `keyword` datatype will let us query for exact values and aggregate, but wouldn't support numeric operations or sorting.

File: schemas/registry.yml (hunk 2 above)
Comment (webmat) [reply schemas/registry.yml]: Let's revisit this later. Resolving.

**Pull request merged**